Source/RapyutaSimulationPlugins/Private/Sensors/RRROS2HitSensorComponent.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020-2023 Rapyuta Robotics Co., Ltd.
+
+#include "Sensors/RRROS2HitSensorComponent.h"
+
+// RapyutaSimulationPlugins
+#include "Core/RRConversionUtils.h"
+#include "Core/RRGeneralUtils.h"
+
+void URRROS2HitSensorComponent::BeginPlay()
+{
+    Super::BeginPlay();
+
+    // default target is owner
+    if (TargetObjects.Num() == 0)
+    {
+        TargetObjects.Add(GetOwner());
+    }
+
+    for (const auto target : TargetObjects)
+    {
+        BindCallback(target);
+    }
+}
+
+void URRROS2HitSensorComponent::BindCallback(UObject* InTargetObject)
+{
+    auto primitiveComp = Cast<UPrimitiveComponent>(InTargetObject);
+    if (primitiveComp)
+    {
+        primitiveComp->OnComponentHit.AddDynamic(this, &URRROS2HitSensorComponent::OnTargetComponentHit);
+        return;
+    }
+
+    auto actor = Cast<AActor>(InTargetObject);
+    if (actor)
+    {
+        actor->OnActorHit.AddDynamic(this, &URRROS2HitSensorComponent::OnTargetActorHit);
+        return;
+    }
+
+    UE_LOG_WITH_INFO_NAMED(LogRapyutaCore, Warning, TEXT("TargetObject must be child class of UPrimitiveComponent or AActor"))
+}
+
+void URRROS2HitSensorComponent::OnTargetComponentHit(UPrimitiveComponent* HitComp,
+                                                     AActor* OtherActor,
+                                                     UPrimitiveComponent* OtherComp,
+                                                     FVector NormalImpulse,
+                                                     const FHitResult& Hit)
+{
+    OnHit(HitComp->GetOwner(), OtherActor, NormalImpulse, Hit, HitComp->GetName());
+}
+
+void URRROS2HitSensorComponent::OnTargetActorHit(AActor* SelfActor,
+                                                 AActor* OtherActor,
+                                                 FVector NormalImpulse,
+                                                 const FHitResult& Hit)
+{
+    OnHit(SelfActor, OtherActor, NormalImpulse, Hit);
+}
+
+void URRROS2HitSensorComponent::OnHit(AActor* SelfActor,
+                                      AActor* OtherActor,
+                                      FVector NormalImpulse,
+                                      const FHitResult& Hit,
+                                      const FString& Name)
+{
+    Data = FROSHitEvent();
+    Data.SelfName = Name.IsEmpty() ? SelfActor->GetName() : Name;
+    Data.OtherActorName = OtherActor->GetName();
+    Data.NormalImpluse = NormalImpulse;
+    Data.HitResult = URRConversionUtils::HitResultUEToROS(Hit);
+    Data.OtherComponentName = Data.HitResult.ComponentName;
+    if (IsIgnore(SelfActor, OtherActor, Hit.Component.Get()))

```

Comment:
A minor detail :
Will the Data var be used if IsIgnore() is true? If not we could move the initialization after